dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java

```diff
@@ -101,15 +101,25 @@
 
     private boolean aggregateable;
 
+    private Class<? extends ValueTypeOptions> valueTypeOptionsClass;
+
     ValueType()
     {
         this.javaClass = null;
     }
 
+    ValueType( Class<?> javaClass, boolean aggregateable, Class<? extends ValueTypeOptions> valueTypeOptionsClass )
+    {
+        this.javaClass = javaClass;
+        this.aggregateable = aggregateable;
+        this.valueTypeOptionsClass = valueTypeOptionsClass;
+    }
+
     ValueType( Class<?> javaClass, boolean aggregateable )
     {
         this.javaClass = javaClass;
         this.aggregateable = aggregateable;
+        this.valueTypeOptionsClass = null;
```

Comment:
Could maybe use `this(...)` to call the constructor with most parameters.

Reply:
Hi, I'm not quite sure what you mean here. Can you give an example maybe?

Reply:
I mean you could call the newly added constructor with `this(javaClass, aggregateable, null);` instead of assigning to fields. 
Usually this is possible and it helps to see quickly that all constructors basically boil down to one which also means the others are sort of the java way of providing default arguments.

Reply:
Ah ok, well. I just changed it from this style :)

Reply:
The argument was that it looks a bit redundant with so many enums with null as last parameter

Reply:
I think what Jan meant was include a call to this() inside the largest constructor:

```
ValueType( Class<?> javaClass, boolean aggregateable, Class<? extends ValueTypeOptions> valueTypeOptionsClass )
    {
        this( javaClass, aggregateable );
        this.valueTypeOptionsClass = valueTypeOptionsClass;
    }
```